plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,798 @@
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class DeleteLicense:
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        license_id = module.params.get('license_id')
+        check_license_id(self, module, license_id, "delete")
+        delete_license_url = f"/redfish/v1/LicenseService/Licenses/{license_id}"
+        delete_license_response = self.idrac.invoke_request(delete_license_url, 'DELETE')
+        status = delete_license_response.status_code
+        if status == 204:
+            module.exit_json(msg=SUCCESS_DELETE_MSG, changed=True)
+        else:
+            module.exit_json(FAILURE_MSG.format(operation="delete", license_id=license_id), failed=True)
+        return delete_license_response
+
+
+class ExportLicense:
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        share_type = module.params.get('share_parameters').get('share_type')
+        license_id = module.params.get('license_id')
+        check_license_id(self, module, license_id, "export")
+        if share_type == "local":
+            export_license_status = self.__export_license_local(module)
+        elif share_type in ["http", "https"]:
+            export_license_status = self.__export_license_http(module)
+            job_status = get_job_status(self, module, export_license_status)
+        elif share_type == "cifs":
+            export_license_status = self.__export_license_cifs(module)
+            job_status = get_job_status(self, module, export_license_status)
+        elif share_type == "nfs":
+            export_license_status = self.__export_license_nfs(module)
+            job_status = get_job_status(self, module, export_license_status)
+        status = export_license_status.status_code
+        if share_type in ["http", "https", "cifs", "nfs"]:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True, job_details=job_status)
+            else:
+                module.exit_json(msg=FAILURE_MSG.format(operation="export", license_id=license_id), failed=True, job_details=job_status)
+        else:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True, job_details={})
+            else:
+                module.exit_json(msg=FAILURE_MSG.format(operation="export", license_id=license_id), failed=True, job_details={})
+        return export_license_status
+
+    def __export_license_local(self, module):
+        uri = validate_and_get_first_resource_id_uri(
+            self.module, self.idrac, SYSTEMS_URI)
+        export_license_url = f"{uri[0]}/{EXPORT_LICENSE_LOCAL}"
+        payload = {}
+        payload["EntitlementID"] = module.params.get('license_id')
```

Comment:
Can we take the value of common parameters like license_id once and use it at other methods as well.

Reply:
No changes done. Will proceed with existing approach